core/snippets/snippets.py

```diff
@@ -23,17 +23,26 @@
     desc="Directory (relative to Talon user) containing additional snippets",
 )
 
-context_map = {
-    # `_` represents the global context, ie snippets available regardless of language
-    "_": Context(),
-}
-snippets_map = {}
+context_map = None
+
+
+def create_context_map():
+    global context_map
+    context_map = {
+        # `_` represents the global context, ie snippets available regardless of language
+        "_": Context(),
+    }
+
+    # Create a context for each defined language
+    for lang in language_ids:
+        ctx = Context()
```

Comment:
If we decide to keep this implementation we definitely should check if the context already exists before creating a new one